internal/condition/eval/eval_test.go

```diff
@@ -90,3 +92,77 @@ condition correct_ip(ip: string) {
 		})
 	}
 }
+
+// TestDefaultCELEvaluationCost is used to ensure we don't decreasee the default evaluation cost
+// of CEL expressions, which would break API compability.
+//
+// Critical paths involving ABAC Condition evaluations use the EvaluateTupleCondition function,
+// and so we test that directly to give us higher confidence we're not introducing a compability
+// issue.
+func TestDefaultCELEvaluationCost(t *testing.T) {
```

Comment:
The goal with this test was not to try and test CEL itself but just to make sure we don't introduce an API incompatibility with CEL evaluation cost enforcement. So the tests included herein are just to give some confidence around that, and not testing CEL directly. However, in order to construct some of these tests you have to have a deeper understanding of how CEL cost calculation works.